### PR TITLE
Run Travis build on multiple Xcode versions

### PR DESCRIPTION
**Pull request by brockboland**
Another test, don't merge

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
+    - osx_image: xcode8.2
+      env: SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SampleProject/SampleProject.xcworkspace"                           SCAN_SCHEME="VOKCoreDataManager-OSX"     SCAN_SDK="macosx"
+    - osx_image: xcode8.1
+      env: SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SwiftSampleProject/SwiftSampleProject.xcworkspace"                 SCAN_SCHEME="SwiftyVokoder"              SCAN_SDK="iphonesimulator"
+    - osx_image: xcode8.2
+      env: SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SwiftSampleProject/SwiftSampleProject.xcworkspace"                 SCAN_SCHEME="SwiftyVokoder"              SCAN_SDK="iphonesimulator"
+    - osx_image: xcode8.1
+      env: SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="Paging Data Source Example/Paging Data Source Example.xcworkspace" SCAN_SCHEME="Paging Data Source Example" SCAN_SDK="iphonesimulator"
+    - osx_image: xcode8.2
+      env: SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="Paging Data Source Example/Paging Data Source Example.xcworkspace" SCAN_SCHEME="Paging Data Source Example" SCAN_SDK="iphonesimulator"
```
Comment (vokal-isaac) [inline .travis.yml]: I think you should be able to do this with
```yaml
matrix:
  osx_image:
    - xcode8.1
    - xcode8.2
  env:
    - SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SampleProject/SampleProject.xcworkspace"                           SCAN_SCHEME="VOKCoreDataManager"         SCAN_SDK="iphonesimulator"
    - SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SampleProject/SampleProject.xcworkspace"                           SCAN_SCHEME="VOKCoreDataManager-tvOS"    SCAN_SDK="appletvsimulator"
    - SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SampleProject/SampleProject.xcworkspace"                           SCAN_SCHEME="VOKCoreDataManager-OSX"     SCAN_SDK="macosx"
    - SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="SwiftSampleProject/SwiftSampleProject.xcworkspace"                 SCAN_SCHEME="SwiftyVokoder"              SCAN_SDK="iphonesimulator"
    - SCAN_SKIP_BUILD=1 SCAN_WORKSPACE="Paging Data Source Example/Paging Data Source Example.xcworkspace" SCAN_SCHEME="Paging Data Source Example" SCAN_SDK="iphonesimulator"
```
but I'm not sure.

Comment (brockboland) [inline .travis.yml]: I'll give it a shot. My impression (based on the documentation) is that `matrix` at the top level can only contain some pre-defined options (`include`, `exclude`, `allow_failures`, `fast_finish`) but it's hard to tell if that's accurate.

Comment (vokal-isaac) [inline .travis.yml]: Yeah, there's definitely ambiguity there.

**Comment by brockboland** [comment]
There's got to be a better way to do this.

Here's what I've tried so far:

- Just make `osx_image` an array: Travis does 5 jobs but doesn't understand `osx_image` array, so nothing runs. (#122)
- Use `matrix: include: osx_image` array: Travis adds 2 jobs (to the regular 5) with no env variables.  (#122)
- `env` array, with `matrix: include: osx_image` array: Same as the previous item (#125)
- Top level `matrix` with `osx_image` and `env` arrays: fails. Travis doesn't undersand it. (#126)
- Set `osx_image` to an array underneatch `matrix`: Travis UI lists 5 jobs, each with `Xcode: [object Object]` (93ceaeb957be6e20226b6f0679296fbd7edd6c6f)
- List every possible matrix option: works (2ad0a861349bb40adfb2014fa38027e24353e225)

The only working solution (so far) is to list all possible `osx_image`/`env` combinations.

**Comment by vokal-isaac** [comment]
Yikes ☹️

**Comment by brockboland** [comment]
If this *does* get merged at some point, make sure to squash the commits

**Comment by vokal-isaac** [comment]
The `osx_image matrix` and reversion could be wiped out with a hard reset and force push, if you wanted.

**Comment by brockboland** [comment]
yeah, but I want to leave the commit there since I mention it in my comment explaining the things I've tried

**Comment by vokal-isaac** [comment]
I'd think squashing would get rid of it, too?

**Comment by brockboland** [comment]
I haven't tried squashing commits in GitHub (and have only done it from the command line a couple times), so I'm not sure. My expectation is that the individual commits would persist in the branch (and this PR), but be combined into a single commit on the master branch.

**Comment by brockboland** [comment]
Now that it's working: starting fresh in #127.

**Pull request closed without merging**